Open Judge System/OJS.Services/Data/Participants/ParticipantsDataService.cs

```diff
@@ -23,5 +24,24 @@ public class ParticipantsDataService : IParticipantsDataService
                 .Where(x => x.Id == participantId)
                 .Select(x => x.IsOfficial)
                 .FirstOrDefault();
+
+        public void ChangeTimeForActiveByContestIdAndMinutes(int contestId, int minutes)
+        {
+            var timeSpan = new TimeSpan(0, 0, minutes, 0);
+
+            var activeParticipants = this.participants
```

Comment:
If a participant has ContestEndTime it is always from online contest, if it is not online contest, `activeParticipants` count will be 0, the the foreach will be skipped and nothing will change.

Reply:
Regardless, you must ensure that the method will not operate for non Online contests, if in the future a new type of contest is created where, the users happen to have ContestEndTimes, or for some other reason a Participant with a ContestEndTime ends up in a normal contest, this will introduce bugs.